### PR TITLE
Introspect the graph to find the db and schema

### LINKED ISSUE #299: add introspection to on run end
### Describe the feature

Add functionality to remove the need for schema/db arguments to the option on run end. We should be able to parse the graph and know the db and schema!

### PR DESCRIPTION
**Pull request by b-per**
This is a:
- [x] bug fix PR with no breaking changes
- [x] new functionality

## Link to Issue 

Closes #299 and updates the README accordingly

## Checklist
- [ ] I have verified that these changes work locally on the following warehouses (Note: it's okay if you do not have access to all warehouses, this helps us understand what has been covered)
    - [ ] BigQuery
    - [ ] Postgres
    - [ ] Redshift
    - [ ] Snowflake
    - [ ] Databricks
    - [x] DuckDB
- [x] I have updated the README.md (if applicable)
- [x] I have added tests & descriptions to my models (and macros if applicable)

### CODE REVIEW

File: macros/on-run-end/print_dbt_project_evaluator_issues.sql [hunk 1]
```diff
@@ -33,11 +33,13 @@
...
-        {% set db_schema = database_package ~ "." ~  schema_package if database_package else schema_package %}
-        {% set db_schema_model = db_schema ~ "." ~  model_checked if db_schema else model_checked %}
+        {% set model_details = graph.nodes.values() | selectattr("name","==",model_checked) | first %}
+        {% set model_schema = model_details.schema %}
+        {% set model_database = model_details.database %}
+        {% set db_schema = model_database ~ "." ~ model_schema if model_database else model_schema %}
 
         {% set sql_statement %}
-        select * from {{ db_schema_model }}
+        select * from {{db_schema}}.{{model_checked}}
```
Comment (dave-connors-3) [inline macros/on-run-end/print_dbt_project_evaluator_issues.sql:42]: worth including the database name here too?

**Review by dave-connors-3 — APPROVED** [review]
seems straightforward to me! small comment on the final select but should be OK

File: macros/on-run-end/print_dbt_project_evaluator_issues.sql (hunk 1 above)
Comment (b-per) [reply macros/on-run-end/print_dbt_project_evaluator_issues.sql:42]: `db_schema` already contains `<optional_database>.schema` if a database is defined. But before we merge this I want to try a new, less hacky, way of getting the table to query, I will move this PR as a draft for now.

**Comment by b-per** [comment]
Updated look with `print_table()`

image

**Comment by fuselessmatt** [comment]
Is this PR fixing the issue in Databricks where stg_nodes contains the wrong catalog/database and schema if you are using Unity Catalog?

**Comment by b-per** [comment]
I don't think that our integration tests have a use case with the Unity Catalog but feel free to report if it works or not!
This is part of the latest 0.5 release.

**Comment by fuselessmatt** [comment]
Sadly it didn't work, but I opened a new issue about it where we can discuss it further

**Pull request merged**